omod/src/main/webapp/resources/scripts/metadata/manageRoles.js

```diff
@@ -122,12 +122,27 @@ angular.module("manageRoles", [ "roleService", "privilegeService", "ngDialog", "
                         $scope.privileges = results;
                         $scope.inheritedPrivilegeFlags = Array($scope.privileges.length);
                         $scope.privilegeFlags = Array($scope.privileges.length);
-                                                
+                        filterPrivileges(); //Do some filtration                       
                         loadInheritedPrivileges();
 
                     });
                 });
             }
+            //Filters Privileges to required criteria. ie Only "Application privileges" are required.
+            function filterPrivileges(){
+            	var inheritedPrivileges =[];
+            	$scope.privileges.forEach(function(val, idx) {
+            		if($scope.privileges[idx].name.indexOf(":")>-1){
+            		var privType = $scope.privileges[idx].name.slice(0,$scope.privileges[idx].name.indexOf(":"));
+            		if(privType == "App"|| privType == "Task"){
+            			inheritedPrivileges.push(val);
+            		}}
+                });   
+            	$scope.privileges = [];//Empty privileges list
```

Comment:
As for some comments in your code, take a look at this: https://dzone.com/articles/5-best-practices-commenting

Reply:
Hahaha I have looked through the resource. Obvious comments aren't required. You know there is this Open Source book I read before were they over advised commenting code for other devs to follow. Note taking in note that not every line code needs a comment. Thanks @dkayiwa . I will become a better me soon..

Reply:
Why don't you just make your code self documenting with well named variables, methods, and other goodies? Definitely if your comments are teaching someone a programming language, they are misplaced. Such belong to computer science text books. 😊 

Commenting should be reserved only for those rare cases where it is necessary. 😊

Reply:
Hahaha okay thanks. @dkayiwa